test/test_autograd.py

```diff
@@ -774,13 +770,13 @@ def bw_hook(grad):
             counter[0] += 1
             return grad * 2
 
-        x = torch.ones(5, 5, requires_grad=True)
+        x = torch.ones(5, 5, dtype=torch.float, requires_grad=True)
         z = bn(x)
         z.register_hook(bw_hook)
         z.sum().backward()
 
         self.assertEqual(counter[0], 1, msg='bw_hook not called')
-        self.assertEqual(x.grad, torch.ones(5, 5) * 2, atol=1e-5, rtol=0)
+        self.assertEqual(x.grad, torch.ones(5, 5, dtype=torch.float) * 2, atol=0.01, rtol=0)
```

Comment:
This should be double and keeping the old tolerance I think.

Reply:
I tried to maintain old tolerance and `dtype=torch.double` here, however when I removed global `torch.set_default_dtype(torch.double)` the `torch.nn.BatchNorm1d` class generates this error:  

```
torch/nn/functional.py", line 2261, in batch_norm
    return torch.batch_norm(
RuntimeError: expected scalar type Double but found Float
```
So I guess in the internal implementation there are some allocation with `dtype=torch.float`. 
I've adjusted the tolerance to a closest value `atol=1e-4, rtol=0`.

Reply:
Ho you need to add a `bn.double()` after creating it to move the Module to the right dtype.